### PR TITLE
add types for react-page-visibility

### PR DESCRIPTION
**Pull request by dannycochran**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by dannycochran** [comment]
cc @pgilad who owns the react-page-visibility repository

File: types/react-page-visibility/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+export interface PageVisibilityProps {
+    children?: React.ReactNode | ((isVisible: boolean, visibilityState: KnownVisibilityStates) => React.ReactNode);
+    onChange?: (isVisible: boolean, visibilityState: KnownVisibilityStates) => void;
+}
+
+/**
+ * A higher order component for managing visibility state via the child
+ * rendering pattern.
+ */
+export class PageVisibility extends Component<PageVisibilityProps> {}
```
Comment (peterblazejewicz) [inline types/react-page-visibility/index.d.ts:27]: this probably should read:
```suggestion
export default class PageVisibility extends Component<PageVisibilityProps> {}
```
https://github.com/pgilad/react-page-visibility/blob/master/src/index.js#L4
so the imports should be amended as well in tests

File: types/react-page-visibility/react-page-visibility-tests.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import { usePageVisibility, PageVisibility, KnownVisibilityStates } from 'react-page-visibility';
```
Comment (peterblazejewicz) [inline types/react-page-visibility/react-page-visibility-tests.tsx:2]:
```suggestion
import PageVisibility, { usePageVisibility, KnownVisibilityStates } from 'react-page-visibility';
```

File: types/react-page-visibility/index.d.ts (hunk 1 above)
Comment (dannycochran) [reply types/react-page-visibility/index.d.ts:27]: ah good catch, thank you @peterblazejewicz

File: types/react-page-visibility/react-page-visibility-tests.tsx (hunk 2 above)
Comment (dannycochran) [reply types/react-page-visibility/react-page-visibility-tests.tsx]: Fixed.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@dannycochran thx!

**Comment by dannycochran** [comment]
Ready to merge

**Pull request merged**